app/controllers/provider_interface/provider_agreements_controller.rb

```diff
@@ -12,11 +12,16 @@ def new_data_sharing_agreement
 
     def create_data_sharing_agreement
       @provider_agreement = ProviderAgreement.new(provider_agreement_params.merge(provider_user: current_provider_user))
-      if @provider_agreement.save
-        redirect_to provider_interface_path
-      else
-        render :data_sharing_agreement
-      end
+
+      render :data_sharing_agreement and return unless @provider_agreement.save
+
+      @provider_setup = ProviderSetup.new(provider_user: current_provider_user)
+      @provider_relationship = @provider_setup.next_relationship_pending
+      @provider_agreement = @provider_setup.next_agreement_pending
```

Comment:
Is `@provider_agreement` needed? `:success` will render only `@provider_agreement` is blank. The redirect will initialise its own variables.

Reply:
:+1: No you're right, this is redundant, I had just been rendering the `:new` template but changed it to a redirect, this is a relic of that earlier approach, will remove

Reply:
@mnacos removed and rebased here https://github.com/DFE-Digital/apply-for-teacher-training/pull/2749/commits/7ffe04852222acbd661ba6aea50b3ac373dc9287